### PR TITLE
vagrant: 2.2.14 -> 2.2.15

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change
ChangeLog: https://github.com/hashicorp/vagrant/blob/v2.2.15/CHANGELOG.md#2215-march-30-2021

Also had to revert a change which improved plugin-errors a bit since it
changed the dependency resolution in a way that broke `vagrant-libvirt`
from `$out/vagrant-plugins/plugins.d`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/vagrant/gemset.nix [hunk 1]
```diff
@@ -363,4 +353,4 @@
     };
     version = "1.3.5";
   };
-}
+}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vagrant/gemset.nix:356]: Missing final new line.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118138` at 13bb7556 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>vagrant</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/tools/vagrant/default.nix:54:5:
```
   |
54 |     ./unofficial-installation-nowarn.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/tools/vagrant/default.nix:56:5:
```
   |
56 |     ./0004-Support-system-installed-plugins.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/tools/vagrant/default.nix:55:5:
```
   |
55 |     ./use-system-bundler-version.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/tools/vagrant/default.nix:57:5:
```
   |
57 |     ./0001-Revert-Merge-pull-request-12225-from-chrisroberts-re.patch
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/vagrant/gemset.nix (hunk 1 above)
Comment (Ma27) [reply pkgs/development/tools/vagrant/gemset.nix:356]: That's a problem of bundix though.

**Comment by Ma27** [comment]
@SuperSandro2000 fixed :)

**Pull request merged**